Kitodo/src/main/java/org/kitodo/production/helper/VariableReplacer.java

```diff
@@ -368,4 +396,121 @@ private String determineReplacementForFirstchild(Matcher variableFinder, String
+     * 
+     * @param stringWithVariables
+     *             string to be checked for file variables.
+     * @return true if string contains file variables.
+     */
+    public boolean containsFiles(String stringWithVariables) {
+        if (Objects.isNull(stringWithVariables)) {
+            return false;
+    	}
+    	return stringWithVariables.contains("(filename)") | stringWithVariables.contains("(filename_without_extension)");
+    }
+
+    /**
+     * Replaces any file variables with the filename passed as string.
+     * Depending on the file variables found, the filename may be written with or without extension.
+     * 
+     * @param stringWithVariables
+     *             string (perhaps) containing file and other placeholders.
+     * @param filename
+     *             string containing filename to replace placeholders.
+     *
+     * @return string in which placeholders have been replaced.
+     */
+    public String replaceWithFilenames(String stringWithVariables, String filename) {
+    	if (Objects.isNull(stringWithVariables)) {
+            return "";
+        }
+
+        stringWithVariables = invokeLegacyVariableReplacer(stringWithVariables);
+
+    	Matcher variableFinder = VARIABLE_FINDER_REGEX.matcher(stringWithVariables);
+    	boolean stringChanged = false;
+    	StringBuffer replacedStringBuffer = null;
+        while (variableFinder.find()) {
+    	    if (!stringChanged) {
+                replacedStringBuffer = new StringBuffer();
+                stringChanged = true;
+            }
+            variableFinder.appendReplacement(replacedStringBuffer, determineReplacement(variableFinder, filename));
+        }
+    	if (stringChanged) {
+            variableFinder.appendTail(replacedStringBuffer);
+            return replacedStringBuffer.toString();
+    	} else {
+            return stringWithVariables;
+    	}
+    }
+
+    /**
+     * If a filename is to be determined, it is determined here.
+     * 
+     * @param variableFinder
+     *            the data structure of the matcher, from which the various
+     *            match groups of the regular expression are read.
+     * @param filename 
+     *            string containing the filename to replace placeholder with.
+     * @return the replacement value, if determinable. If the variable is
+     *         invalid, the string found is returned and no replacement takes
+     *         place. If the variable cannot be replaced because the requested
+     *         data cannot be accessed, the empty string is returned.
+     */
+    private String determineReplacementForFilePlaceholder(Matcher variableFinder, String filename) {
+        switch (variableFinder.group(6)) {
+    	    case "filename":
+                return determineReplacementForFilename(variableFinder, filename);
+    	    case "filename_without_extension":
+                return determineReplacementForFilenameWithoutExtension(variableFinder, filename);
```

Comment:
You can simplify this a lot (and get rid of two more functions) by using `org.apache.commons.io.FilenameUtils`:
```java
switch (variableFinder.group(6)) {
    case "filename":
        return variableFinder.group(1) + FilenameUtils.getName(filename);
    case "basename":
        return variableFinder.group(1) + FilenameUtils.getBaseName(filename);
    default:
        ...
```